xl/xenops_server_xenlight.ml

```diff
@@ -2256,9 +2256,11 @@ module VM = struct
 						xs.Xs.setperms ent rwperm
 					)
 				) (
-					let dev_kinds = [ "vbd"; "vif"; "vfb"; "vkb"; "vfs"; "pci" ] in
 					[ "device"; "error"; "drivers"; "control"; "attr"; "data"; "messages"; "vm-data" ]
-					@ List.map (fun dev_kind -> "device/"^dev_kind) dev_kinds
+					@ if hvm then
+						let dev_kinds = [ "vbd"; "vif"; "vfb"; "vkb"; "vfs"; "pci" ] in
+						List.map (fun dev_kind -> "device/"^dev_kind) dev_kinds
+					else []
```

Comment:
We only need VIF and VBD keys here for the Windows PV drivers. It could be one of the more "exotic" keys (PCI maybe?). Ring0 couldn't extract enough information from the guest to work out what it was tripping over. Perhaps we should consider only creating these. Or, maybe even better, have the Windows PV drivers create these at start of day given we've already created the `.../device` dir with RW permissions for the guest.